src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmListIdService.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.boxc.migration.cdm.services;
+
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+
+import edu.unc.lib.boxc.common.util.URIUtil;
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+
+/**
+ * Service for listing CDM object IDs
+ * @author krwong
+ */
+public class CdmListIdService {
+    private static final Logger log = getLogger(CdmListIdService.class);
+
+    private CloseableHttpClient httpClient;
+    private String cdmBaseUri;
+
+    private int pageSize = 1000; //set default page size
+
+    /**
+     * Get the total number of object IDs for the given collection
+     * @param project
+     * @throw IOException
+     * @return
+     */
+    private int getTotalObjects(MigrationProject project) throws IOException {
+        String collectionId = project.getProjectProperties().getCdmCollectionId();
+        String totalObjectsUrl = "dmwebservices/index.php?q=dmQuery/" + collectionId
```

Comment:
It would make sense to add a constant to this class like sharon suggests:
`public static final String CDM_QUERY_BASE = "dmwebservices/index.php?q=dmQuery/";`
Which could be used for this class and the tests. 

For tests where you are having to construct a lot of urls, you could make another constant that used the first constant, so for example in CdmListIdServiceTest you could have
`public static final String CDM_BASE_URL = "http://example.com:88/dmwebservices/index.php?q=" + CDM_QUERY_BASE +  "gilmer/0/dmrecord/dmrecord/";`
And then when you need to build a url for testing you'd be able to shorten it to
```
assertUrlsCalled(CDM_BASE_URL + "1/0/1/0/0/0/0/json", ...)
```
May help with readability given all the repetitive URLs as well, since only a small part of the url ever changes.